typed/helpers.go

```diff
@@ -28,7 +28,7 @@ import (
 
 // ValidationError reports an error about a particular field
 type ValidationError struct {
-	Path         fieldpath.Path
+	Path         string
```

Comment:
Why? Is this because errors outlive other things? Maybe it's OK to explicitly make a copy for the errors?

Reply:
Yes, the problem is that: 
1. Ideally we don't have a lot of errors, technically we're not supposed to have any when we're doing Updates
2. This forces the escapes of numbers, strings, Values, everything on every single field of objects, for ... a non-nominal use-case.

Reply:
And eventually we just convert everything to a string anyway, so we don't really need the PE

Reply:
But we shouldn't actually construct one of these unless there actually is an error? Why would this cause stuff to escape all the time? It shouldn't be a problem if it escapes only when there's an error?

I guess I'd have expected a []string for the replacement?

Reply:
> Why would this cause stuff to escape all the time? It shouldn't be a problem if it escapes only when there's an error?

Good questions. Unfortunately, It did escape in my tests none the less :-/

> I guess I'd have expected a []string for the replacement?

Does it make a difference? It will be concatenated anyway.

Reply:
I guess I don't actually care how you store it, if a string is best for you that's fine.

I do care that the path always corresponds to exact place in the object that sourced the error, for that I think you need to capture whatever it is you want to store exactly where the error is generated and not build it after the fact with WithPrefix calls :/ The other way to get the guarantee is to exhaustively test every error with its path, which I don't think is easier :/